doc/source/user/how-to-partition.rst

```diff
@@ -0,0 +1,270 @@
+.. _how-to-partition:
+
+=====================================
+How to partition a domain using NumPy
+=====================================
+
+There are a few NumPy functions that are similar in application, but which
+provide slightly different results, which may cause confusion if one is not sure
+when and how to use them. The following guide aims to list these functions and
+describe their recommended usage.
+
+The functions mentioned here are
+
+* `numpy.linspace`
+* `numpy.arange`
+* `numpy.geomspace`
+* `numpy.logspace`
+* `numpy.meshgrid`
+* `numpy.mgrid`
+* `numpy.ogrid`
+
+1D domains (intervals)
+======================
+
+``linspace`` vs. ``arange``
+---------------------------
+
+Both `numpy.linspace` and `numpy.arange` provide ways to partition an interval
+(a 1D domain) into equal-length subintervals. These partitions will vary
+depending on the chosen starting and ending points, and the **step** (the length 
+of the subintervals).
+
+* Use `numpy.arange` if you want integer steps.
+
+  `numpy.arange` relies on step size to determine how many elements are in the
+  returned array, which excludes the endpoint. This is determined through the
+  ``step`` argument to ``arange``.
+
+  Example::
+
+    >>> np.arange(0, 10, 2)  # np.arange(start, stop, step)
+    array([0, 2, 4, 6, 8])
+
+  The arguments ``start`` and ``stop`` should be integer or real, but not
+  complex numbers.
+
+* Use `numpy.linspace` if you want the endpoint to be included in the result, or
+  if you are using a non-integer step size.
```

Comment:
A bit of "why" is missing, maybe something like
```suggestion
  if you are using a non-integer step size since floating-point inaccuracies
  can make ``arange`` results with floating-point numbers confusing.
```

Reply:
Ahh, you have this below. Maybe a "(see below)" instead?

Reply:
Included this in the arange explanation, let me know if it looks good.